internal/observability/stream.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package observability
+
+import (
+	"context"
+
+	"go.uber.org/yarpc/api/transport"
+)
+
+const (
+	_successfulStreamReceive = "Successfully received stream message"
+	_successfulStreamSend    = "Successfully sent stream message"
+	_errorStreamReceive      = "Error receiving stream message"
+	_errorStreamSend         = "Error sending stream message"
+)
+
+var _ transport.StreamCloser = (*streamWrapper)(nil)
+
+type streamWrapper struct {
```

Comment:
should we call this streamMetricsWrapper? 
I think it has 2 parts:
- metrics emission and delegate to the underlying StreamCloser
- disable (by making it a no-op) Close

Reply:
I'd like to refrain from using `metrics` since we also add logging in #1770